jenkins/bootstrap.py

```diff
@@ -735,11 +737,16 @@ def job_args(args):
     return [os.path.expandvars(a) for a in args]
 
 
-def job_script(job, use_json):
+def job_script(job, use_json, jobs_dir):
     """Return path to script for job."""
+    jobs_dir_resolver = lambda f: test_infra('jobs/%s' % f)
+
+    if jobs_dir:
+        jobs_dir_resolver = lambda f: os.path.join(jobs_dir, f)
+
     if not use_json:
```

Comment:
hmm, so as Seb mentioned we use the shell script code path, and this change allows us to put those jobs scripts in each respective repo. (e.g., istio/pilot has prow jobs that live in istio/pilot/prow/ directory). 

It appears that scenarios just reference shell scripts? Additionally -- these scenarios live in kubernetes test infra repo?

I'm trying to prevent Istio from maintaining a separate `bootstrap.py` by introducing flags that allow us to run it. I want to work toward that, do you think this makes sense Erick/what's the way forward?